test/e2e/framework.go

```diff
@@ -611,6 +617,138 @@ func validatePodIP(podNetworkCIDR, podIP string) (bool, error) {
+			},
+		},
+		RestartPolicy: v1.RestartPolicyNever,
+	}
+	if nodeName != "" {
+		podSpec.NodeSelector = map[string]string{
+			"kubernetes.io/hostname": nodeName,
+		}
+	}
+	if nodeName == masterNodeName() {
+		// tolerate NoSchedule taint if we want Pod to run on master node
+		noScheduleToleration := v1.Toleration{
+			Key:      "node-role.kubernetes.io/master",
+			Operator: v1.TolerationOpExists,
+			Effect:   v1.TaintEffectNoSchedule,
+		}
+		podSpec.Tolerations = []v1.Toleration{noScheduleToleration}
+	}
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+			Labels: map[string]string{
+				"pod-name": podName,
+				"app":      "nginx",
+			},
+		},
+		Spec: podSpec,
+	}
+	if _, err := data.clientset.CoreV1().Pods(testNamespace).Create(pod); err != nil {
+		return err
+	}
+	return nil
+}
+
+// createNginxService exposes nginx as a service.
+func (data *TestData) createNginxService(podName string, svcName string) error {
+	port := 80
+	_, err := data.clientset.CoreV1().Services(testNamespace).Create(&v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: svcName,
+			Labels: map[string]string{
+				"app": "nginx",
+			},
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{{
+				Port:       int32(port),
+				TargetPort: intstr.IntOrString{Type: 0, IntVal: int32(port)},
+			}},
+			Selector: map[string]string{
+				"app": "nginx",
+			},
+		},
+	})
+	return err
+}
+
+// deleteNginxService deletes the nginx service
+func (data *TestData) deleteNginxService(svcName string) error {
+	if err := data.clientset.CoreV1().Services(testNamespace).Delete(svcName, nil); err != nil {
+		return fmt.Errorf("unable to cleanup svc %v: %v", svcName, err)
+	}
+	return nil
+}
+
+// createNetworkPolicyToTestIPBlockWithExcept creates a network policy with IPBlock's Except .
+// 192.168.1.2 ~ 192.168.1.255 allowd
+// 192.168.1.0 ~ 192.168.1.1 rejected.
+func (data *TestData) createNetworkPolicyToTestIPBlockWithExcept(networkPolicyName string, exceptIP string) (*networkingv1.NetworkPolicy, error) {
+	policy := &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: networkPolicyName,
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "nginx",
+				},
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{
+				{
+					From: []networkingv1.NetworkPolicyPeer{
+						{
+							IPBlock: &networkingv1.IPBlock{
+								CIDR: clusterInfo.podNetworkCIDR,
+								Except: []string{
+									exceptIP + "/32",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	return data.clientset.NetworkingV1().NetworkPolicies(testNamespace).Create(policy)
+}
+
+// networkPolicyWaitForExisting polls the K8s apiserver until the specified network policy is found (in the test Namespace).
+func (data *TestData) networkPolicyWaitForExisting(timeout time.Duration, name string) (*networkingv1.NetworkPolicy, error) {
```

Comment:
why is this needed? I think APIServer is synchronous when creating objects

Reply:
Well, I think it may take some time to create because I observe that without waiting, chances are that the testcase fails at Pod0's wget behavior which should succeed.

Reply:
The failures should be not related to the waiting, because the creation of API object is synchronous so the get query will always success, it may be only because the get query takes some time to finish so leave some space for the asynchronous network policy enforcement. 
But I think the pod should be able to connect service both before applying the policy and after that if you are creating policy allows them. Have you created other policies denying the traffic?